src/kinetics/InterfaceKinetics.cpp

```diff
@@ -595,4 +595,30 @@ void InterfaceKinetics::setPhaseStability(const size_t iphase, const int isStabl
     }
 }
 
+double InterfaceKinetics::InterfaceCurrent(const size_t iphase)
+{
+    int nSp = thermo(iphase).nSpecies();
+    double charge_k[nSp];
+    double sdot_k[nSp];
+    doublereal netProdRates[m_kk];
+
+    thermo(iphase).getCharges(charge_k);
+
+    getNetProductionRates(netProdRates);
+
+    for(int k=0; k<nSp; k++)
+    {
+        sdot_k[k] = netProdRates[m_start[iphase]+k];
+    }
+
+    double dotProduct = 0.0;
+
+    for(int k=0; k<nSp; k++)
+    {
+        dotProduct += charge_k[k]*sdot_k[k];
+    }
```

Comment:
those two loops can be combined, which eliminates the need for `s_dot_k`. As a minor formatting quibble, there should be spaces before and after operators.

Reply:
Removed the `s_dot_k` variable and integrated both loops into one single loop.